spec/config-spec.coffee

```diff
@@ -1106,6 +1106,44 @@ describe "Config", ->
         expect(atom.config.get('foo.bar.str', scope: ['.source.js'])).toBe 'omg'
         expect(atom.config.get('foo.bar.str', scope: ['.source.coffee'])).toBe 'ok'
 
+      describe 'when a schema is added after config values have been set', ->
+        schema = null
+        beforeEach ->
+          schema =
+            type: 'object'
+            properties:
+              int:
+                type: 'integer'
+                default: 2
+
+        it 'the values set respect the new schema', ->
+          expect(atom.config.set('foo.bar.int', 'nope')).toBe true
```

Comment:
So if you have something like `foo.bar.str` set to `'hey'`, does it get unset after the schema gets set even though it doesn't have a schema?

Mind want to add some coverage for this case.